File: Secure Code Assessment/assets/vulnerable_sample.py
import os
import sqlite3
import hashlib
# pyrefly: ignore [missing-import]
from flask import Flask, request, render_template_string

app = Flask(__name__)

# HARDCODED SENSITIVE CREDENTIALS
AWS_SECRET_KEY = "AKIAIOSFODNN7EXAMPLE_SECRET_KEY_HERE"
DB_PASSWORD = "SuperSecretPassword123!"
JWT_SECRET = "my_super_secret_jwt_token_key"

@app.route('/login', methods=['POST'])
def login():
    username = request.form.get('username')
    password = request.form.get('password')
    
    # WEAK AUTHENTICATION & HASHING (MD5)
    hashed_pass = hashlib.md5(password.encode()).hexdigest()
    
    # SQL INJECTION VULNERABILITY
    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    query = f"SELECT * FROM users WHERE username = '{username}' AND password = '{hashed_pass}'"
    cursor.execute(query) # Dangerous raw SQL query concatenation
    user = cursor.fetchone()
    
    return "Logged in successfully" if user else "Invalid credentials"

@app.route('/ping', methods=['GET'])
def ping():
    host = request.args.get('host')
    
    # COMMAND INJECTION VULNERABILITY
    command = f"ping -c 1 {host}"
    os.system(command) # Unchecked command execution
    return f"Pinged {host}"

@app.route('/greet')
def greet():
    name = request.args.get('name', 'Guest')
    
    # CROSS-SITE SCRIPTING (XSS)
    template = f"<h1>Welcome, {name}!</h1>"
    return render_template_string(template) # Unsanitized user input in template

@app.route('/read_file')
def read_file():
    filename = request.args.get('file')
    
    # INSECURE FILE HANDLING / PATH TRAVERSAL
    file_path = f"/var/www/uploads/{filename}"
    with open(file_path, 'r') as f: # User input directly used in open()
        content = f.read()
    return content

if __name__ == '__main__':
    app.run(debug=True) # DEBUG MODE ENABLED IN PRODUCTION

File: Secure Code Assessment/assets/vulnerable_sample.php
<?php

// HARDCODED CREDENTIALS
$db_host = "127.0.0.1";
$db_user = "root";
$db_pass = "SuperSecretPHPPass2026";
$secret_jwt_key = "a9f87b6c5d4e3f2a1";

// SQL INJECTION
if (isset($_GET['user_id'])) {
    $id = $_GET['user_id'];
    $conn = mysqli_connect($db_host, $db_user, $db_pass, "app_db");
    // Concatenating raw GET parameter directly into query
    $result = mysqli_query($conn, "SELECT * FROM users WHERE id = " . $id);
}

// CROSS-SITE SCRIPTING (XSS)
if (isset($_POST['comment'])) {
    // Printing unsanitized HTML output directly to page
    echo "<div class='comment'>User says: " . $_POST['comment'] . "</div>";
}

// COMMAND INJECTION & DANGEROUS EXEC
if (isset($_GET['ip'])) {
    $ip = $_GET['ip'];
    // System execution of unsanitized IP parameter
    $output = passthru("ping -c 4 " . $ip);
    system("traceroute " . $ip);
}

// INSECURE FILE HANDLING / PATH TRAVERSAL
if (isset($_GET['page'])) {
    $page = $_GET['page'];
    // Including arbitrary file based on URL parameter
    include("pages/" . $page);
    echo file_get_contents("/var/www/data/" . $page);
}

// WEAK AUTHENTICATION & CRYPTOGRAPHY
function storeUserPassword($password) {
    // Insecure MD5 algorithm without salt
    $hashed = md5($password);
    return $hashed;
}

// DANGEROUS DESERIALIZATION
if (isset($_COOKIE['session_data'])) {
    // Unserialize untrusted cookie data
    $user_session = unserialize($_COOKIE['session_data']);
}

?>
